service/isiService.go

```diff
@@ -563,14 +565,34 @@ func (svc *isiService) AddExportClientNetworkIdentifierByIDWithZone(ctx context.
 	}
 
 	log.Debugf("AddExportClientNetworkIdentifierByID adding '%s' as client to export id '%d'", clientToUse, exportID)
-	if err = addClientFunc(ctx, exportID, accessZone, clientToUse); err == nil {
+
+	// Reading ignoreHosts value as bool
+	ignoreHosts, err1 := strconv.ParseBool(os.Getenv("X_CSI_ISI_IGNORE_HOSTS"))
```

Comment:
rename variable name err1 to err

Reply:
you are already reading this value when the serivce starts in service.go@initializeServiceOpts(), so no need to read it again. Also this value can be set to each array(present in the secret) independently similar to isi path. So you need to introduce a new param in secret.